asv_bench/benchmarks/rolling.py

```diff
@@ -225,6 +225,19 @@ def time_rolling_offset(self, method):
         getattr(self.groupby_roll_offset, method)()
 
 
+class Groupby2:
```

Comment:
how long does this take on prior branch? we want this < 1s

Reply:
only around 150ms, and now down to 20ms with this branch (the relative difference is smaller as in the original reported case, because the overhead of iterating and creating tuples gets larger for for larger frames)

Reply:
ok great.